submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/SourceGenerator.kt

```diff
@@ -5,23 +5,25 @@ import ebi.ac.uk.io.sources.ComposedFileSource
 import ebi.ac.uk.io.sources.FilesSource
 import ebi.ac.uk.io.sources.ListFilesSource
 import ebi.ac.uk.io.sources.PathFilesSource
-import ebi.ac.uk.security.integration.model.api.GroupMagicFolder
 import ebi.ac.uk.security.integration.model.api.SecurityUser
 import java.io.File
 
-@Suppress("SpreadOperator")
 class SourceGenerator {
-    fun getSubmissionSources(user: SecurityUser, files: List<File>, rootPath: String) =
-        ComposedFileSource(userFiles(user, rootPath), requestFiles(files), *getGroupSources(user.groupsFolders))
+    fun getSubmissionSources(user: SecurityUser, files: List<File>, rootPath: String): FilesSource {
+        val sources = mutableListOf<FilesSource>()
```

Comment:
What about the submission folder?

Reply:
as we do not know at this level if exists (submission may be new) it is added dynamically by submitter